src/briefcase/platforms/macOS/__init__.py

```diff
@@ -160,37 +181,64 @@ def select_identity(self, identity=None):
             print()
             selection = select_option(identities, input=self.input)
             identity = identities[selection]
-            print("selected", identity)
 
         return identity
 
-    def sign(self, path, entitlements, identity):
+    def sign(self, path, identity, entitlements=None):
         """
         Code sign a file.
 
         :param path: The path to the file to sign.
-        :param entitlements: The path to the entitlements file to use.
         :param identity: The code signing identity to use. Either the 40-digit
             hex checksum, or the string name of the identity.
+        :param entitlements: The path to the entitlements file to use.
         """
+        options = 'runtime' if identity != '-' else None
+        process_command = [
+            'codesign',
+            os.fsdecode(path),
+            '--sign', identity,
+            '--force',
+        ]
+        if entitlements:
+            process_command.append('--entitlements')
+            process_command.append(os.fsdecode(entitlements))
+        if options:
+            process_command.append('--options')
+            process_command.append(options)
+
         try:
-            print("Signing", path)
+            print(f"Signing {Path(path).relative_to(self.base_path)}")
             self.subprocess.run(
-                [
-                    'codesign',
-                    '--sign', identity,
-                    '--entitlements', os.fsdecode(entitlements),
-                    '--deep', os.fsdecode(path),
-                    '--force',
-                    '--options', 'runtime',
-                ],
+                process_command,
+                stderr=subprocess.PIPE,
                 check=True,
             )
-        except subprocess.CalledProcessError:
-            print()
-            raise BriefcaseCommandError(
-                "Unable to code sign {path}.".format(path=path)
-            )
+        except subprocess.CalledProcessError as e:
+            errors = e.stderr.decode('utf-8', errors='replace')
+            if 'not signed at all' in errors:
+                print("... file requires a deep sign; retrying")
+                try:
+                    self.subprocess.run(
+                        process_command + ['--deep'],
+                        stderr=subprocess.PIPE,
+                        check=True,
+                    )
+                except subprocess.CalledProcessError:
+                    raise BriefcaseCommandError(f"Unable to deep code sign {path}.")
+
+            elif any(
+                msg in errors
+                for msg in [
+                    'unsupported format for signature',
```

Comment:
@dreavjr Similarly - how do you reproduce this message?

Reply:
This message happens when we try to sign a file that is not executable code (meaning that we had a false positive on the filtering procedures that select which files are going to be signed). This may happen, for example, if some package uses the ".so" extension for some proprietary data format.

That happened to me on Qt for Python 6 because they had a couple of non-executable object files marked with the execution permission flag.

Reply:
Are you saying you get the "unsupported format" message on a vanilla Pyside6 project? I've just generated a "HelloWorld" project with the PySide6 template, and the only file in the bundle that needed special handling was `PySide6/Qt/lib/QtWebEngineCore.framework/Versions/A/QtWebEngineCore` - and that needed a `--deep` pass. 

I've tried:
* Creating a text file with a `.so` extension
* Creating a text file with a `.dylib` extension
* Creating a text file with no extension
* Creating all of the above, and marking them executable
* Creating a binary file with Mach-O magic, but garbage content after that, marked as executable

and passing them to `codesign`, and I don't get any errors.

Reply:
When it happened to me it was due to a `.o` object file (wrongly?) marked as executable in PySide6. I can look for it as soon as I get access to my main computer. It is conceivable, though, that the behavior is dependent on codesign version. I'll try to reproduce the error on my side and let you know exactly which version I'm using.

Reply:
@freakboy3742 I'm no longer being able to reproduce a situation that triggers this message. It's possible that it became redundant with the other checks I added upstream. Maybe we can safely remove this downstream check?

Reply:
Thanks - I guess it doesn't hurt to retain it just in case. I've flagged the test case to say "don't know how you generate it", so if anyone ever works it out, we can clarify.